File: TP/TP03/soma.c
#include <stdio.h>
int somadig(int n){
	if(n<10) return n;
	else return n%10 + somadig(n/10);
}

int main(){
	int n;
	while(scanf("%d",&n)){
		int soma = somadig(n);
		printf("%d\n",soma);
	}

return 0;
}

File: AED2/TP/TP03/Soma.java
import java.util.Scanner;

public class Soma {
    public static int SomaDig(int n){
        if(n<10) return n;
    	else return n%10 + SomaDig(n/10);
    }

    public static void main(String []args){
        Scanner sc = new Scanner(System.in);
        
        while (sc.hasNextInt()) {
            int n = sc.nextInt();
            int soma = SomaDig(n);
            System.out.println(soma);
        }
    }
}